src/server/common/controller/question-page-controller.js

```diff
@@ -77,8 +78,13 @@ export class QuestionPageController {
       ...req.yar.get(this.page.sectionKey),
       [this.page.questionKey]: answer.toState()
     })
-    return h.redirect(
-      calculateNextPage(payload.nextPage, this.page.nextPage(answer).urlPath)
-    )
+
+    const nextPage = this.page.nextPage(answer)
+
+    if (nextPage instanceof ExitPage) {
```

Comment:
Should this not be also part of the calculateNextPage function?

Reply:
But that page doesn't (yet) use the concept of a Page